### PR TITLE
ISSUE-950 [ICS Reply] CalendarEvent/accept JMAP method

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/950

### CODE REVIEW

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventAcceptMethod.scala [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        .fold(_ => SMono.error[Calendar](InvalidCalendarFileException(blobId)), result => SMono.just(result)))
+      .flatMap(calendar => validate(calendar)
+        .fold(e => SMono.error[Calendar](InvalidCalendarFileException(blobId, e.getMessage)), SMono.just))
+  }
+
+  private def validate(calendar: Calendar): Either[IllegalArgumentException, Calendar] =
+    if (calendar.getComponents("VEVENT").isEmpty) {
+      Left(new IllegalArgumentException("The calendar file must contain VEVENT componennt"))
+    } else if (calendar.getMethod.getValue != "REQUEST") {
+      Left(new IllegalArgumentException("The calendar must is REQUEST method"))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventAcceptMethod.scala:137]: The calendar must have REQUEST a s a method

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventAcceptMethod.scala [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        .disposition("attachment"))
+  }
+
+  private def extractRecipient(calendar: Calendar): MailAddress =
+    calendar.getComponents[VEvent]("VEVENT").asScala.headOption
+      .flatMap(CalendarOrganizerField.from)
+      .flatMap(_.mailto) match {
+      case Some(value) => value
+      case None => throw new IllegalArgumentException("Cannot extract the organizer from the calendar event.")
+    }
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventAcceptMethod.scala:174]: Use Either

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventAcceptMethod.scala [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        case e => SMono.error(e)
+      })
+
+  private def getLanguageLocale(request: CalendarEventReplyRequest): Locale = request.language.map(_.language).getOrElse(LANGUAGE_DEFAULT)
+
+  private def enqueue(mail: Mail): Unit = try {
+    queue.enQueue(mail)
+  } finally {
+    LifecycleUtil.dispose(mail)
+  }
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/CalendarEventAcceptMethod.scala:121]: Enqueue reactive

**Comment by vttranlina** [comment]
Almost completed
TODO: add Distributed test & review more test cases & clean code

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAcceptMethodContract.scala [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+        s"""[
+           |    "CalendarEvent/accept",
+           |    {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "accepted": [ "$blobId" ]
+           |    },
+           |    "c1"
+           |]""".stripMargin)
+  }
+  @Test
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAcceptMethodContract.scala:96]:
```suggestion
  }
  
  @Test
```

**Comment by vttranlina** [comment]
Manual test with thunderbird

https://github.com/linagora/tmail-backend/assets/81145350/596b7b5b-80cb-40fa-b97e-cd53057c5587

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]
<3

**Comment by Arsnael** [comment]
Can rebase

**Comment by Arsnael** [comment]
@vttranlina : Thing I just encountered: don't forget that if the account accepting the event is not part of the attendee list in the ics file, it will reject the action. Which is the case in your integration tests it seems (bob@domain.tld not present as an attendee in your ics resource files)

**Review by chibenwa — APPROVED** [review]

**Pull request merged**